### PR TITLE
random: provide range functionality

### PR DESCRIPTION
**Pull request by authmillenon**
Provides a function to generate a random value in an arbitrary range.

### CODE REVIEW

**Comment by Lotterleben** [comment]
ACK :)

File: sys/include/random.h [hunk 1]
```diff
@@ -54,6 +54,20 @@ void genrand_init_by_array(uint32_t init_key[], int key_length);
  */
 uint32_t genrand_uint32(void);
 
+/**
+ * @brief   generates a random number r with a < r <= b.
```
Comment (Kijewski) [inline sys/include/random.h]: `a <= r < b`

Comment (authmillenon) [inline sys/include/random.h]: Arghs… didn't see that before merging -.-

Comment (authmillenon) [inline sys/include/random.h]: Fixed in #3441

Comment (authmillenon) [inline sys/include/random.h]: I meant #3442

**Pull request merged**